test/Driver/link-time-opt.swift

```diff
@@ -0,0 +1,99 @@
+// CHECK-SIMPLE-FULL-DAG: -emit-bc
+// CHECK-SIMPLE-FULL-DAG: -lto=llvm-full
+// CHECK-SIMPLE-FULL-DAG: -o [[BITCODEFILE:.*\.bc]]
+
+// CHECK-SIMPLE-FULL-windows-msvc: clang
+// CHECK-SIMPLE-FULL-windows-msvc-DAG: -fuse-ld=lld
+// CHECK-SIMPLE-FULL-windows-msvc-DAG: -flto=full
+// CHECK-SIMPLE-FULL-windows-msvc-DAG: [[BITCODEFILE]]
+
+// CHECK-SIMPLE-FULL-linux-gnu: clang
+// CHECK-SIMPLE-FULL-linux-gnu-DAG: -flto=full
+// CHECK-SIMPLE-FULL-linux-gnu-DAG: -fuse-ld=lld
+// CHECK-SIMPLE-FULL-linux-gnu-DAG: [[BITCODEFILE]]
+// CHECK-SIMPLE-FULL-linux-gnu-NOT: swift-autolink-extract
+
+
+
+// RUN: %swiftc_driver -driver-print-jobs %S/../Inputs/empty.swift -lto=llvm-thin -static -emit-library -target x86_64-apple-macosx10.9  | %FileCheck %s --check-prefix=CHECK-STATIC-LIB-THIN --check-prefix=CHECK-STATIC-LIB-THIN-macosx 
+// RUN: %swiftc_driver -driver-print-jobs %S/../Inputs/empty.swift -lto=llvm-thin -static -emit-library -target x86_64-unknown-linux-gnu | %FileCheck %s --check-prefix=CHECK-STATIC-LIB-THIN --check-prefix=CHECK-STATIC-LIB-THIN-linux-gnu
+
+// CHECK-STATIC-LIB-THIN: swift
+// CHECK-STATIC-LIB-THIN-DAG: -emit-bc
+// CHECK-STATIC-LIB-THIN-DAG: -lto=llvm-thin
+// CHECK-STATIC-LIB-THIN-DAG: -o [[BITCODEFILE:.*\.bc]]
+
+// CHECK-STATIC-LIB-THIN-macosx: libtool
+// CHECK-STATIC-LIB-THIN-macosx-DAG: [[BITCODEFILE]]
+
+// CHECK-STATIC-LIB-THIN-linux-gnu: llvm-ar
+// CHECK-STATIC-LIB-THIN-linux-gnu-DAG: [[BITCODEFILE]]
+// CHECK-STATIC-LIB-THIN-linux-gnu-NOT: swift-autolink-extract
+
+
+
+// RUN: %swiftc_driver -driver-print-jobs %S/../Inputs/empty.swift -lto=llvm-full -static -emit-library -target x86_64-apple-macosx10.9  | %FileCheck %s --check-prefix=CHECK-STATIC-LIB-FULL --check-prefix=CHECK-STATIC-LIB-FULL-macosx 
+// RUN: %swiftc_driver -driver-print-jobs %S/../Inputs/empty.swift -lto=llvm-full -static -emit-library -target x86_64-unknown-linux-gnu | %FileCheck %s --check-prefix=CHECK-STATIC-LIB-FULL --check-prefix=CHECK-STATIC-LIB-FULL-linux-gnu
+
+// CHECK-STATIC-LIB-FULL: swift
+// CHECK-STATIC-LIB-FULL-DAG: -emit-bc
+// CHECK-STATIC-LIB-FULL-DAG: -lto=llvm-full
+// CHECK-STATIC-LIB-FULL-DAG: -o [[BITCODEFILE:.*\.bc]]
+
+// CHECK-STATIC-LIB-FULL-macosx: libtool
+// CHECK-STATIC-LIB-FULL-macosx-DAG: [[BITCODEFILE]]
+
+// CHECK-STATIC-LIB-FULL-linux-gnu: llvm-ar
+// CHECK-STATIC-LIB-FULL-linux-gnu-DAG: [[BITCODEFILE]]
+// CHECK-STATIC-LIB-FULL-linux-gnu-NOT: swift-autolink-extract
+
+
+// Ensure that filelist accepts bitcode files produced by frontend
+
+// RUN: %empty-directory(%t)
+// RUN: echo "int dummy;" >%t/a.cpp
+// RUN: cc -c %t/a.cpp -o %t/a.o
+// RUN: %swiftc_driver -save-temps -driver-print-jobs %S/../Inputs/empty.swift %t/a.o -lto=llvm-full -target x86_64-apple-macosx10.9 -driver-filelist-threshold=0 -o filelist 2>&1 | tee %t/forFilelistCapture | %FileCheck -check-prefix FILELIST %s
+// RUN: tail -2 %t/forFilelistCapture | head -1 | sed 's/.*-output-filelist //' | sed 's/ .*//' > %t/output-filelist
```

Comment:
I want an end to end test. Maybe in the interpreter dir. With something that LTO can hit for sure.

Reply:
Added dead symbol strip e2e test case